src/main/java/org/apache/gossip/manager/GossipManager.java

```diff
@@ -207,11 +208,12 @@ public void shutdown() {
     gossipServiceRunning.set(false);
     gossipThreadExecutor.shutdown();
     gossipCore.shutdown();
-    if (passiveGossipThread != null) {
-      passiveGossipThread.shutdown();
-    }
-    if (activeGossipThread != null) {
-      activeGossipThread.shutdown();
+    if(null != this.bundle) {
```

Comment:
Again I would lose the interface and and manage these as references to ActiveGossipThread and PassiveGossipThread